src/Control/Tracer.hs

```diff
@@ -59,11 +59,14 @@ module Control.Tracer
     , stdoutTracer
     , debugTracer
     -- * Transforming tracers
+    , mapTracer
```

Comment:
@avieth do you think that `mapTracer` can be implemented in the arrow based tracer? If not we should not export it here, as we are moving towards it.

Reply:
`mapTracer` is a bit too powerful to work with the arrow based tracer, since it's factored into two parts: emitting and non-emitting. The argument to `mapTracer` doesn't carry that information, and so it's impossible make a sensible definition.

But `traceTraversable` _can_ be defined for the arrow tracer, because `traverse` is sufficiently polymorphic

```Haskell
Kleisli . traverse . runKleisli :: (Traversable t, Applicative m) => Kleisli m a b -> Kleisli m (t a) (t b)
```

and so the same transformation can be applied to both the emitting and non-emitting parts of the tracer arrow

```Haskell
traceTraversable :: (Applicative m, Traversable t) => TracerA m a b -> TracerA m (t a) (t b)
traceTraversable (Squelching k) = Squelching $ Kleisli $ \ta -> traverse (runKleisli k) ta
traceTraversable (Emitting emittingPart nonEmittingPart) = Emitting
  (Kleisli (traverse (runKleisli emittingPart))) (Kleisli (traverse (runKleisli nonEmittingPart)))
```

I think `traceTraversable` could be the most general thing of this sort that we'd even want from the arrow tracer.

Reply:
Thanks @avieth, this makes sense.

@lemastero could you just remove the `mapTracer` from the export list.

Reply:
Removed `mapTracer` from export https://github.com/input-output-hk/contra-tracer/pull/3/commits/bb7aa431afd299bc421c8331abb6926275afadc9